packages/test/local-server-stress-tests/src/stressDataObject.ts

```diff
@@ -0,0 +1,320 @@
+	type: "createDataStore";
+	asChild: boolean;
+	tag: `datastore-${number}`;
+}
+
+export interface CreateChannel {
+	type: "createChannel";
+	channelType: string;
+	tag: `channel-${number}`;
+}
+
+export type StressDataObjectOperations = UploadBlob | CreateDataStore | CreateChannel;
+
+export class StressDataObject extends DataObject {
+	public static readonly factory = new Lazy(() => {
+		const factory = new DataObjectFactory(
+			"StressDataObject",
+			StressDataObject,
+			[...ddsModelMap.values()].map((v) => v.factory),
+			{},
+			[["StressDataObject", new LazyPromise(() => factory)]],
+		);
+		return factory;
+	});
+
+	get StressDataObject() {
+		return this;
+	}
+
+	private defaultStressObject: DefaultStressDataObject = makeUnreachableCodePathProxy(
+		"defaultStressDataObject",
+	);
+	protected async getDefaultStressDataObject(): Promise<DefaultStressDataObject> {
+		const defaultDataStore =
+			await this.context.containerRuntime.getAliasedDataStoreEntryPoint("default");
+		assert(defaultDataStore !== undefined, "default must exist");
+
+		const maybe: FluidObject<DefaultStressDataObject> | undefined =
+			await defaultDataStore.get();
+		assert(maybe.DefaultStressDataObject !== undefined, "must be DefaultStressDataObject");
+		return maybe.DefaultStressDataObject;
+	}
+
+	/**
+	 * this map is special, and doesn't participate in stress. it hold data
+	 * about the name of channels which have been created. these created channel
+	 * may or may not be attached and be available
+	 */
+	private channelNameMap: ISharedMap = makeUnreachableCodePathProxy("channelNameMap");
+	protected async initializingFirstTime(props?: any): Promise<void> {
+		this.channelNameMap = SharedMap.create(this.runtime, "channelNameMap");
+		this.channelNameMap.bindToContext();
+		this.channelNameMap.set("root", this.root.attributes.type);
+	}
+
+	public async getChannels() {
+		const channels: IChannel[] = [];
+		for (const [name] of this.channelNameMap.entries()) {
+			// similar to container objects, the entries in this map
+			// can appear before the underlying channel is attached,
+			// so getting the channel can fail, and we need to try
+			// to get all channel each time, as we have no way to
+			// observer when a channel moves from detached to attached,
+			// especially on remove clients/
+			const channel = await this.runtime.getChannel(name).catch(() => undefined);
+			if (channel !== undefined) {
+				channels.push(channel);
+			}
+		}
+		return channels;
+	}
+
+	protected async hasInitialized(): Promise<void> {
+		this.defaultStressObject = await this.getDefaultStressDataObject();
+
+		this.channelNameMap = (await this.runtime.getChannel(
+			"channelNameMap",
+		)) as any as ISharedMap;
+	}
+
+	public get attached() {
+		return this.runtime.attachState === AttachState.Attached;
+	}
+
+	public uploadBlob(tag: `blob-${number}`, contents: string) {
+		void this.runtime.uploadBlob(stringToBuffer(contents, "utf-8")).then((handle) =>
```

Comment:
Why are these methods sync and voiding their impls?  Feels like someone further up the stack should be making that call?